packages/@aws-cdk/aws-kinesisfirehose/lib/delivery-stream.ts

```diff
@@ -339,15 +353,29 @@ export class DeliveryStream extends DeliveryStreamBase {
      */
     encryptionKey?.grantEncryptDecrypt(role);
 
+    const sourceStreamConfig = props.sourceStream ? {
+      kinesisStreamArn: props.sourceStream.streamArn,
+      roleArn: role.roleArn,
+    } : undefined;
+    const readStreamGrant = props.sourceStream?.grantRead(role);
+    if (readStreamGrant && readStreamGrant.principalStatement) {
+      readStreamGrant.principalStatement.addActions('kinesis:DescribeStream');
```

Comment:
Can you add a comment about why this is necessary? Looks like this fixes the issue described in  https://github.com/aws/aws-cdk/issues/10783?